### PR TITLE
To Do list: Microverse

### PR DESCRIPTION
**Pull request by iwillteixeira**
### To Do list: add & remove

- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for editing task descriptions.
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for the "Clear all completed" button (use filter() method).
- Index set to the value of the new array length.
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- All changes to the To-Do List should be saved in local storage.

### CODE REVIEW

File: src/jsmodules/array.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const firstDisplay = [
-  { task: 'Wash the dishes', bool: true, id: 1 },
-  { task: 'Build a To-Do app', bool: true, id: 2 },
-  { task: 'dont\' forget about the webpack', bool: true, id: 3 },
+  { task: 'Wash the dishes', bool: true, id: 0 },
+  { task: 'Build a To-Do app', bool: true, id: 1 },
+  { task: 'dont\' forget about the webpack', bool: true, id: 2 },
```
Comment (Lameck1) [inline src/jsmodules/array.js]: - [x] I suggest you remove these hardcoded items as required of us in the project requirements. This `firstDisplay` variable has been imported in the `index.js` file, which means you'll have to take care of removing the importation too. The project requirement for this is quoted below:

> **- Remove all hardcoded items from the tasks array.**

File: src/index.js [hunk 2]
```diff
@@ -1,22 +1,69 @@
 import '../node_modules/bootstrap/dist/js/bootstrap.bundle.js';
 import './scss/style.scss';
 import firstDisplay from './jsmodules/array.js';
```
Comment (Lameck1) [inline src/index.js]: - [REMINDER] Remember to remove this importation and refactor any part of code that uses it.

File: src/jsmodules/add.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { uPStoreValues } from './storevalues.js';
+import { ul } from './selectors.js';
+
+export class TaskAdd {
+  constructor(task, bool, id) {
+    this.task = task;
+    this.bool = bool;
+    this.id = id;
```
Comment (Lameck1) [inline src/jsmodules/add.js]: - [x] Let's talk about these two task properties.
* Firstly, a more semantic way to call this property defined as `bool` would be `completed` just like in the requirements. `bool` is too generic and does not give enough meaning as per the context of a task. Think about `task.completed` VS `task.bool`, which one sounds better? consider renaming it to **`completed`**.
* Secondly, a task item should also have the property of `index` which I assume you've defined as `id`. I suggest you rename this one too.

Let us take a look at the requirement below:

> **- By default new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).**

File: src/jsmodules/add.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  }
+}
+
+export const addItem = (item, array) => {
+  const sizeArr = JSON.parse(localStorage.getItem('taskList')).length;
+  const itemAdd = new TaskAdd(item, true, sizeArr);
```
Comment (Lameck1) [inline src/jsmodules/add.js]: - [x] Let's take a look at the screenshot below:
Todos

Take a keen look at the `id` property that I've suggested you rename to `index`. Referring back to the requirement quoted in the above review thread, the value of this property should be the value of the **`NEW`** **_array.length_**. Mark the word NEW. **_This means that if I add a task for the first time, my new array will have a length of one, and therefore the index property of my first task will be one._**

In the requirement, we have an example stating that, if you're adding a 5th task to the list, the index of that task should equal to 5. But, the fifth task in the screenshot above has an `id` value of 4.

## Suggestion
In your logic simply add (+ 1) to the length of the array. This means you `refresh` method will need to refactored as suggested in the review thread below:

File: src/jsmodules/storevalues.js [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
 export const uPStoreValues = (e) => {
   localStorage.setItem('taskList', JSON.stringify(e));
 };
+
+export const refresh = (array) => {
+  array.forEach((e, a) => {
+    e.id = a;
+  });
+  uPStoreValues(array);
+};
```
Comment (Lameck1) [inline src/jsmodules/storevalues.js]:
```suggestion
export const refresh = (array) => {
  array.forEach((e, a) => {
    e.id = a + 1;
  });
  uPStoreValues(array);
};
```

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @iwillteixeira, 

Well done for the good job you've done so far 👏🏼. What an effort 👍🏼!
Let's just work on a few adjustments to help complete this project.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/jsmodules/array.js (hunk 1 above)
Comment (iwillteixeira) [reply src/jsmodules/array.js]: Removed all hardcoded items from the task array.

File: src/index.js (hunk 2 above)
Comment (iwillteixeira) [reply src/index.js]: Thanks for the reminder, I removed them all.

File: src/jsmodules/add.js (hunk 3 above)
Comment (iwillteixeira) [reply src/jsmodules/add.js]: I renamed all the variables.

File: src/jsmodules/add.js (hunk 4 above)
Comment (iwillteixeira) [reply src/jsmodules/add.js]: done

File: src/jsmodules/storevalues.js (hunk 5 above)
Comment (iwillteixeira) [reply src/jsmodules/storevalues.js]: done

**Review by mikethreels — APPROVED** [review]
Hi Will, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**